### PR TITLE
Legacy to Haru 11

### PR DESCRIPTION
**Pull request by MikeZeDev**
* Migudm
* Mode Scanlator
* Momo no Hana Scan
* Monochrome Scans

### CODE REVIEW

File: web/src/engine/websites/MonochromeScans.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            const mangas = await this.getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+
+    }
+
+    private async getMangasFromPage(page: number, provider: MangaPlugin) {
+        const uri = new URL('/manga', apiUrl);
+        uri.searchParams.set('limit', '50');
```
Comment (ronny1982) [inline web/src/engine/websites/MonochromeScans.ts:46]: Is `50` the maximum or can it be increased further?

File: web/src/engine/websites/MonochromeScans.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        uri.searchParams.set('offset', String(page - 1 * 50));
```
Comment (ronny1982) [inline web/src/engine/websites/MonochromeScans.ts:47]: Is this by intention?
`7 - 1 * 50` => `-43`
or should it be
`(7 - 1) * 50` => `300`

File: web/src/engine/websites/MonochromeScans.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        const request = new FetchRequest(uri.href);
+        const data = await FetchJSON<APIManga[]>(request);
+        return data.map(entry => new Manga(this, provider, entry.id, entry.title));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const request = new FetchRequest(new URL('/manga/'+manga.Identifier + '/chapters', apiUrl).href);
+        const data = await FetchJSON<APIChapter[]>(request);
+        return data.map(chapter => {
+            let title = '';
```
Comment (ronny1982) [inline web/src/engine/websites/MonochromeScans.ts:57]: Handling spaces in conditional composed titles is quite cumbersome.
Suggestion to utilize arrays:
```typescript
const title = [
    chapter.volume ? `Volume ${chapter.volume}` : undefined,
    'Chapter',
    chapter.number,
    chapter.name ? '-' ? undefined,
    chapter.name
].filter(entry => entry).join(' ');

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/MonochromeScans.ts (hunk 2 above)
Comment (MikeZeDev) [reply web/src/engine/websites/MonochromeScans.ts:47]: IIRC, Linter said parenthesis are useless. I'll have a look.

File: web/src/engine/websites/MonochromeScans.ts (hunk 1 above)
Comment (MikeZeDev) [reply web/src/engine/websites/MonochromeScans.ts:46]: It seems so, by default its 10, and more than 50 triggers errors 422.

Comment (MikeZeDev) [reply web/src/engine/websites/MonochromeScans.ts:48]: image

**Comment by MikeZeDev** [comment]
They changed the structure of JSON for getting manga list, i've adjusted it while we are at it :)

**Pull request merged**